docs/source/api/apollo-server.md

```diff
@@ -343,15 +343,44 @@ addMockFunctionsToSchema({
    to standard error. Specify this function to process errors in a different
    way.
 
-*  `privateVariables`: Array<String> | boolean
+*  `privateVariables`: Array<String\> | boolean
 
+   DEPRECATING IN VERSION XX.XX.XX for `sendVariableValues`, which will support the same
+   functionalities but allow for more flexibility.
+   
    A case-sensitive list of names of variables whose values should not be sent
    to Apollo servers, or 'true' to leave out all variables. In the former
    case, the report will indicate that each private variable was redacted in
    the latter case, no variables are sent at all.
+   
+* `sendVariableValues`: { valueModifier: (options: { variables: Record<string, any>, operationString?: string } ) => Record<string, any> }
+                     | { exceptVariableNames: Array<String\> }
+                     | { safelistAll: boolean }
```

Comment:
maybe even `{sendAll: true} | {sendNone: true}`? Where the latter is the default.  (You can declare in TypeScript that only `true` is allowed somewhere!)